sim-lib/src/lib.rs

```diff
@@ -129,6 +130,52 @@ pub struct SimParams {
     pub activity: Vec<ActivityParser>,
 }
 
+/// Either a value or a range parsed from the simulation file.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum ValueOrRange<T> {
+    Value(T),
+    Range(T, T),
+}
+
+impl<T> ValueOrRange<T>
+where
+    T: std::cmp::PartialOrd + rand_distr::uniform::SampleUniform + Copy,
+{
+    /// Get the enclosed value. If value is defined as a range, sample from it uniformly at random.
+    pub fn value(&self) -> T {
+        match self {
+            ValueOrRange::Value(x) => *x,
+            ValueOrRange::Range(x, y) => {
+                let mut rng = rand::thread_rng();
+                rng.gen_range(*x..*y)
+            },
+        }
+    }
+
+    /// Whether this is a range or not
+    pub fn is_range(&self) -> bool {
```

Comment:
nit: I believe this is an unused function?

Reply:
Oh yeah, this was part of the previous approach: https://github.com/bitcoin-dev-project/sim-ln/blob/cc19a99e50459cd1e1ba790df3e74fd44b2af66b/sim-lib/src/lib.rs#L829-L837

Will remove it